src/modules/python/Makefile.am

```diff
@@ -39,14 +39,79 @@
 
 #
 
+pbs_v1_sodir = $(libdir)/python/altair
+
+pbs_v1_so_LTLIBRARIES = \
+	_pbs_v1.la \
+	_pbs_ifl.la
+
+nodist_pbs_v1_so_PYTHON = $(top_builddir)/src/lib/Libifl/pbs_ifl.py
+
+_pbs_v1_la_SOURCES = \
+	$(top_builddir)/src/lib/Libattr/attr_fn_acl.c \
+	$(top_builddir)/src/lib/Libattr/attr_fn_entlim.c \
+	$(top_builddir)/src/lib/Libattr/attr_fn_resc.c \
+	$(top_builddir)/src/lib/Libattr/attr_fn_time.c \
+	$(top_builddir)/src/lib/Libattr/attr_node_func.c \
+	$(top_builddir)/src/lib/Libattr/job_attr_def.c \
+	$(top_builddir)/src/lib/Libattr/node_attr_def.c \
+	$(top_builddir)/src/lib/Libattr/queue_attr_def.c \
+	$(top_builddir)/src/lib/Libattr/resc_def_all.c \
+	$(top_builddir)/src/lib/Libattr/resv_attr_def.c \
+	$(top_builddir)/src/lib/Libattr/sched_attr_def.c \
+	$(top_builddir)/src/lib/Libattr/svr_attr_def.c \
+	$(top_srcdir)/src/lib/Libpython/shared_python_utils.c \
+	$(top_srcdir)/src/lib/Libpython/common_python_utils.c \
+	$(top_srcdir)/src/lib/Libpython/pbs_python_external.c \
+	$(top_srcdir)/src/lib/Libpython/pbs_python_svr_external.c \
+	$(top_srcdir)/src/lib/Libpython/module_pbs_v1.c \
+	$(top_srcdir)/src/lib/Libpython/pbs_python_svr_internal.c \
+	$(top_srcdir)/src/lib/Libpython/pbs_python_svr_size_type.c \
+	$(top_srcdir)/src/lib/Libpython/pbs_python_import_types.c \
+	$(top_srcdir)/src/lib/Libutil/entlim.c \
+	$(top_srcdir)/src/lib/Libutil/hook.c \
+	$(top_srcdir)/src/lib/Libutil/work_task.c \
+	$(top_srcdir)/src/server/svr_attr.c \
+	$(top_srcdir)/src/server/resc_attr.c \
+	$(top_srcdir)/src/server/setup_resc.c \
+	$(top_srcdir)/src/server/vnparse.c \
+	pbs_v1_module_init.c
+_pbs_v1_la_CPPFLAGS = \
+	-I$(top_srcdir)/src/include \
+	@PYTHON_INCLUDES@ \
+	@KRB5_CFLAGS@
+_pbs_v1_la_LDFLAGS = \
+	-module \
+	-shared \
+	-export-dynamic \
+	-avoid-version \
+	@PYTHON_LDFLAGS@
+_pbs_v1_la_LIBADD = \
+	$(top_builddir)/src/lib/Libpbs/libpbs.la \
+	@PYTHON_LIBS@
+
+nodist__pbs_ifl_la_SOURCES = $(top_builddir)/src/lib/Libifl/pbs_ifl_wrap.c
+_pbs_ifl_la_CPPFLAGS = \
+	-I$(top_srcdir)/src/include \
+	@PYTHON_INCLUDES@
+_pbs_ifl_la_LDFLAGS = \
+	-module \
+	-shared \
+	-export-dynamic \
+	-avoid-version \
+	-no-undefined \
+	@PYTHON_LDFLAGS@
+_pbs_ifl_la_LIBADD = \
+	$(top_builddir)/src/lib/Libpbs/libpbs.la \
+	@PYTHON_LIBS@
+
```

Comment:
I tried compiling your code, but was getting an error under src/modules/python:
```
src/modules/python> make
/bin/sh ../../../libtool  --tag=CC   --mode=link gcc  -g -Wall -Werror -Wno-unused-result -fPIC -D_LARGEFILE_SOURCE -O2     -module -shared -export-dynamic -avoid-version -L/home/bayucan/2021.1.0/2020.1.1.20200901012250/linux312_x86_64-work/python/lib/python3.6/config-3.6m-x86_64-linux-gnu -L/home/bayucan/2021.1.0/2020.1.1.20200901012250/linux312_x86_64-work/python/lib -lpython3.6m -lpthread -ldl  -lutil -lm  -Xlinker -export-dynamic -L/home/bayucan/2021.1.0/2020.1.1.20200901012250/linux312_x86_64-work/tcltk/lib -L/home/bayucan/2021.1.0/2020.1.1.20200901012250/linux312_x86_64-work/alsdk/lib/linux_x64 -o _pbs_v1.la -rpath /opt/pbs/lib/python/altair _pbs_v1_la-attr_fn_acl.lo _pbs_v1_la-attr_fn_entlim.lo _pbs_v1_la-attr_fn_resc.lo _pbs_v1_la-attr_fn_time.lo _pbs_v1_la-attr_node_func.lo _pbs_v1_la-job_attr_def.lo _pbs_v1_la-node_attr_def.lo _pbs_v1_la-queue_attr_def.lo _pbs_v1_la-resc_def_all.lo _pbs_v1_la-resv_attr_def.lo _pbs_v1_la-sched_attr_def.lo _pbs_v1_la-svr_attr_def.lo _pbs_v1_la-shared_python_utils.lo _pbs_v1_la-common_python_utils.lo _pbs_v1_la-pbs_python_external.lo _pbs_v1_la-pbs_python_svr_external.lo _pbs_v1_la-module_pbs_v1.lo _pbs_v1_la-pbs_python_svr_internal.lo _pbs_v1_la-pbs_python_svr_size_type.lo _pbs_v1_la-pbs_python_import_types.lo _pbs_v1_la-entlim.lo _pbs_v1_la-hook.lo _pbs_v1_la-work_task.lo _pbs_v1_la-svr_attr.lo _pbs_v1_la-resc_attr.lo _pbs_v1_la-setup_resc.lo _pbs_v1_la-vnparse.lo _pbs_v1_la-pbs_v1_module_init.lo ../../../src/lib/Libpbs/libpbs.la -lpython3.6m -lpthread -ldl  -lutil -lm  -ldl -lcrypt -lc -lc 
libtool: link: gcc -shared  -fPIC -DPIC  .libs/_pbs_v1_la-attr_fn_acl.o .libs/_pbs_v1_la-attr_fn_entlim.o .libs/_pbs_v1_la-attr_fn_resc.o .libs/_pbs_v1_la-attr_fn_time.o .libs/_pbs_v1_la-attr_node_func.o .libs/_pbs_v1_la-job_attr_def.o .libs/_pbs_v1_la-node_attr_def.o .libs/_pbs_v1_la-queue_attr_def.o .libs/_pbs_v1_la-resc_def_all.o .libs/_pbs_v1_la-resv_attr_def.o .libs/_pbs_v1_la-sched_attr_def.o .libs/_pbs_v1_la-svr_attr_def.o .libs/_pbs_v1_la-shared_python_utils.o .libs/_pbs_v1_la-common_python_utils.o .libs/_pbs_v1_la-pbs_python_external.o .libs/_pbs_v1_la-pbs_python_svr_external.o .libs/_pbs_v1_la-module_pbs_v1.o .libs/_pbs_v1_la-pbs_python_svr_internal.o .libs/_pbs_v1_la-pbs_python_svr_size_type.o .libs/_pbs_v1_la-pbs_python_import_types.o .libs/_pbs_v1_la-entlim.o .libs/_pbs_v1_la-hook.o .libs/_pbs_v1_la-work_task.o .libs/_pbs_v1_la-svr_attr.o .libs/_pbs_v1_la-resc_attr.o .libs/_pbs_v1_la-setup_resc.o .libs/_pbs_v1_la-vnparse.o .libs/_pbs_v1_la-pbs_v1_module_init.o   -Wl,-rpath -Wl,/home/bayucan/2021.1.0/2020.1.1.20200901012250/linux312_x86_64-work.orig/src/lib/Libpbs/.libs -Wl,-rpath -Wl,/opt/pbs/lib -L/home/bayucan/2021.1.0/2020.1.1.20200901012250/linux312_x86_64-work/python/lib/python3.6/config-3.6m-x86_64-linux-gnu -L/home/bayucan/2021.1.0/2020.1.1.20200901012250/linux312_x86_64-work/python/lib -L/home/bayucan/2021.1.0/2020.1.1.20200901012250/linux312_x86_64-work/tcltk/lib -L/home/bayucan/2021.1.0/2020.1.1.20200901012250/linux312_x86_64-work/alsdk/lib/linux_x64 ../../../src/lib/Libpbs/.libs/libpbs.so -lz -lcrypto -lpython3.6m -lpthread -lutil -lm -ldl -lcrypt -lc  -O2 -Wl,-export-dynamic   -Wl,-soname -Wl,_pbs_v1.so -o .libs/_pbs_v1.so
/usr/lib64/gcc/x86_64-suse-linux/4.8/../../../../x86_64-suse-linux/bin/ld: /home/bayucan/2021.1.0/2020.1.1.20200901012250/linux312_x86_64-work/python/lib/python3.6/config-3.6m-x86_64-linux-gnu/libpython3.6m.a(abstract.o): relocation R_X86_64_32S against `_Py_NotImplementedStruct' can not be used when making a shared object; recompile with -fPIC
/home/bayucan/2021.1.0/2020.1.1.20200901012250/linux312_x86_64-work/python/lib/python3.6/config-3.6m-x86_64-linux-gnu/libpython3.6m.a: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
Makefile:665: recipe for target '_pbs_v1.la' failed
make: *** [_pbs_v1.la] Error 1
```
I was able to get around this with this Makefile.am fix:
```
diff -c Makefile.am.orig Makefile.am
*** Makefile.am.orig    2020-09-14 13:30:35.169228291 -0700
--- Makefile.am 2020-09-14 14:42:46.649020870 -0700
***************
*** 84,94 ****
        -module \
        -shared \
        -export-dynamic \
!       -avoid-version \
!       @PYTHON_LDFLAGS@
  _pbs_v1_la_LIBADD = \
!       $(top_builddir)/src/lib/Libpbs/libpbs.la \
!       @PYTHON_LIBS@
  
  nodist__pbs_ifl_la_SOURCES = $(top_builddir)/src/lib/Libifl/pbs_ifl_wrap.c
  _pbs_ifl_la_CPPFLAGS = \
--- 84,92 ----
        -module \
        -shared \
        -export-dynamic \
!       -avoid-version
  _pbs_v1_la_LIBADD = \
!       $(top_builddir)/src/lib/Libpbs/libpbs.la
  
  nodist__pbs_ifl_la_SOURCES = $(top_builddir)/src/lib/Libifl/pbs_ifl_wrap.c
  _pbs_ifl_la_CPPFLAGS = \
***************
*** 99,109 ****
        -shared \
        -export-dynamic \
        -avoid-version \
!       -no-undefined \
!       @PYTHON_LDFLAGS@
  _pbs_ifl_la_LIBADD = \
!       $(top_builddir)/src/lib/Libpbs/libpbs.la \
!       @PYTHON_LIBS@
  
  pbsmoduledir = $(libdir)/python/altair/pbs
  
--- 97,105 ----
        -shared \
        -export-dynamic \
        -avoid-version \
!       -no-undefined
  _pbs_ifl_la_LIBADD = \
!       $(top_builddir)/src/lib/Libpbs/libpbs.la
  
  pbsmoduledir = $(libdir)/python/altair/pbs
  
```

Reply:
Any information you can provide to replicate the link error would be greatly appreciated. 
 Please let me know which version of autoconf, automake and libtool you were using.  Also, were those versions of the autotools the default installation for a specific OS and version?  If so, which OS and version?

Reply:
I compiled it under SLES12, GNU Autoconf 2.69, libtool.m4 (macro_version=2.4.2
macro_revision=1.3337), automake (GNU automake) 1.13.4

Reply:
and linked against Python 3.6.

Reply:
@bayucan Much thanks for the info and the patch.  After some additional testing, it appears that you are correct in that those flags and libraries are not required.  I'm not entirely sure why given that I obviously added them for a reason, but I'm guessing I introduced them during the dev process to eliminate some error, and the changes weren't ultimately necessary and were resolved by another change.  I've applied and committed the changes you proposed, test them locally, and they appear to have passed CI testing.  Again, thanks for both the feedback and the patch.